site-src/geps/gep-1426.md

```diff
@@ -146,6 +141,23 @@ This should be enforced by controller validation, which should set an `Accepted:
 
 The logic described here is a bit verbose, but could be validated with conformance tests, and is intended to allow an `HTTPRoute` configuring mesh traffic rules to additionally specify a `Gateway` `parentRef` (or `HTTPRoute` `parentRef` if [GEP-1058: Route Inclusion and Delegation](https://github.com/kubernetes-sigs/gateway-api/pull/1085) moves forward) to apply the same routing rules when exposing a mesh service outside the cluster.
 
+### Namespace boundaries
+
+In a mesh, routes can be configured by two personas:
+
+* Service producers, who want to modify behavior of *inbound* requests to their services
+* Service consumers, who want to modify behavior of *outbound* requests to other services.
+
+While these concepts are not directly exposed in the API, a route is implicitly fulfilling one of these rolls and behaves differently depending on the role.
+
+A route is a producer route when the `parentRef` refers to a service in the *same namespace*. This route SHOULD apply to all incoming requests to the service, including from clients in other namespaces.
```

Comment:
What happens when both a consumer and producer route apply to the same Service?

Reply:
Added a (lot) of discussion about this. A bit worried I am creeping the scope of this PR, but its all connected so has to be done

Reply:
Compositing is still difficult with 1 impl, as doing the merge is quite complex. I take it that is how its done in consul today, right?

One thing I based this on is that Gateway route+producer route should behave like consumer route+producer route. So if we are doing merging for consumer routes, we ought to do it for Gateway, which brings its own can of worms.

I wrote up a bit on this a long time ago, might be helpful to guide discussion:
[Public Gateway API Mesh Route Merging (1).pdf](https://github.com/kubernetes-sigs/gateway-api/files/9923961/Public.Gateway.API.Mesh.Route.Merging.1.pdf)

Reply:
I think the thread starting with https://github.com/kubernetes-sigs/gateway-api/pull/1493/files#r1009709390 more completely covers this, so I commented there.